pkg/machines/scripts/install_machine.sh

```diff
@@ -89,10 +89,24 @@ fi
 # backup
 DOMAIN_FILE="`mktemp`"
 
-virsh -c "$CONNECTION_URI" -q destroy "$VM_NAME" 2>/dev/null
+virsh -c "$CONNECTION_URI" -q destroy "$VM_NAME" 2>/dev/null || true
 virsh -c "$CONNECTION_URI" -q dumpxml "$VM_NAME" > "$DOMAIN_FILE"
 virsh -c "$CONNECTION_URI" -q undefine "$VM_NAME" --managed-save
 
+handleFailure() {
+    #inject metadata, and define
+    sed "$METADATA_LINE""i $METADATA" "$DOMAIN_FILE" | virsh -c "$CONNECTION_URI" -q define /dev/stdin
+    rm -f "$DOMAIN_FILE"
+    # return back if failed
+    if vmExists "$VM_NAME"; then
+        # undefine if the domain was created but still failed
+        virsh -c "$CONNECTION_URI" -q undefine "$VM_NAME" --managed-save
+        virsh -c "$CONNECTION_URI" -q define "$DOMAIN_FILE"
```

Comment:
could either of these, or the `virsh define` above , fail? Then they need to be `|| true`'d, or handleFailure should run `set +e`.

Reply:
You added `set +x` now, but I gues this is a typo -- your script never does `set -x`, so there is no debug tracing to disable. Supposedly you meant `-e`? (Burning the midnight oil.. :grin: )